src/coreclr/gc/handletable.cpp

```diff
@@ -540,22 +540,13 @@ void HndLogSetEvent(OBJECTHANDLE handle, _UNCHECKED_OBJECTREF value)
 
 #ifndef DACCESS_COMPILE
 /*
- * HndWriteBarrier
+ * HndWriteBarrierWorker
  *
  * Resets the generation number for the handle's clump to zero.
  *
  */
-void HndWriteBarrier(OBJECTHANDLE handle, OBJECTREF objref)
+void HndWriteBarrierWorker(OBJECTHANDLE handle, _UNCHECKED_OBJECTREF value)
 {
-    STATIC_CONTRACT_NOTHROW;
-    STATIC_CONTRACT_GC_NOTRIGGER;
-    STATIC_CONTRACT_MODE_COOPERATIVE;
-
-    // unwrap the objectref we were given
-    _UNCHECKED_OBJECTREF value = OBJECTREF_TO_UNCHECKED_OBJECTREF(objref);
-
-    _ASSERTE (objref != NULL);
```

Comment:
This assert should probably not be just deleted, let's keep it here (but test value parameter, of course).

Reply:
ahh that's a mistake..I didn't mean to; good catch! will add it back.

Reply:
I'm actually doing an assert in both HndWriteBarrier and HndWriteBarrierWorker because the validation on an OBJREF actually calls ValidateObject but it won't do that on a _UNCHECKED_OBJECTREF.